core/src/main/java/com/linecorp/armeria/client/retry/RetryRuleBuilder.java

```diff
@@ -0,0 +1,285 @@
+     */
+    public RetryRuleBuilder onStatus(Predicate<? super HttpStatus> statusFilter) {
+        requireNonNull(statusFilter, "statusFilter");
+        if (this.statusFilter != null) {
+            this.statusFilter = this.statusFilter.or(statusFilter);
+        } else {
+            @SuppressWarnings("unchecked")
+            final Predicate<HttpStatus> cast = (Predicate<HttpStatus>) statusFilter;
+            this.statusFilter = cast;
+        }
+        return this;
+    }
+
+    /**
+     * Adds the specified exception type for a {@link RetryRule} which will retry
+     * if an {@link Exception} is raised and that is instance of the specified {@code exception}.
+     */
+    public RetryRuleBuilder onException(Class<? extends Throwable> exception) {
+        requireNonNull(exception, "exception");
+        return onException(exception::isInstance);
+    }
+
+    /**
+     * Adds the specified {@code exceptionFilter} for a {@link RetryRule} which will retry
+     * if an {@link Exception} is raised and the specified {@code exceptionFilter} returns {@code true}.
+     */
+    public RetryRuleBuilder onException(Predicate<? super Throwable> exceptionFilter) {
+        requireNonNull(exceptionFilter, "exceptionFilter");
+        if (this.exceptionFilter != null) {
+            this.exceptionFilter = this.exceptionFilter.or(exceptionFilter);
+        } else {
+            @SuppressWarnings("unchecked")
+            final Predicate<Throwable> cast = (Predicate<Throwable>) exceptionFilter;
+            this.exceptionFilter = cast;
+        }
+        return this;
+    }
+
+    /**
+     * Makes a {@link RetryRule} retry on any {@link Exception}.
+     */
+    public RetryRuleBuilder onException() {
+        return onException(unused -> true);
+    }
+
+    /**
+     * Makes a {@link RetryRule} retry on an {@link UnprocessedRequestException} which means that the request
+     * has not been processed by the server. Therefore, you can safely retry the request without worrying about
+     * the idempotency of the request.
+     */
+    public RetryRuleBuilder onUnprocessed() {
+        return onException(UnprocessedRequestException.class);
+    }
+
+    /**
+     * Sets the {@linkplain Backoff#ofDefault() default backoff} and returns a newly created {@link RetryRule}.
+     */
+    public RetryRule thenBackoff() {
+        return thenBackoff(Backoff.ofDefault());
+    }
+
+    /**
+     * Sets the specified {@link Backoff} and returns a newly created {@link RetryRule}.
+     */
+    public RetryRule thenBackoff(Backoff backoff) {
+        requireNonNull(backoff, "backoff");
+        return build(RetryRuleDecision.retry(backoff));
+    }
+
+    /**
+     * Sets a {@link Backoff} which limits the number of attempts up to the specified value and
+     * never waits between attempts. Returns a newly created {@link RetryRule}.
+     */
+    public RetryRule thenImmediately(int maxAttempts) {
```

Comment:
Having a retrying client without backoff is such a bad practice in almost every situation that recommend not having a shortcut, if someone really wants this bad behavior they can pass spend the couple of lines of code to pass `Backoff` to `thenBackoff`

Reply:
This method is dangerous in most cases. Let's remove this shortcut. 👍